### PR TITLE
Home view

### PR DESCRIPTION
**Pull request by jakubjen**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -45,8 +53,15 @@ export function HttpLoaderFactory(http: HttpClient) {
 			maxAge: 25,
 			logOnly: environment.production,
 		}),
+		AngularFireModule.initializeApp(environment.firebase),
+		AngularFireAnalyticsModule,
+		AngularFirestoreModule,
+		AngularFireAnalyticsModule,
+		AngularFireAuthModule,
+		// provideAnalytics(() => getAnalytics()),
+		// provideAuth(() => getAuth()),
```
Comment (danieljanusm) [inline src/app/app.module.ts:62]: commented code

File: src/app/recipes/recipes-cards/recipes-cards.component.html [hunk 2]
```diff
@@ -10,3 +10,11 @@ <h2 class="m-b-4">{{ 'Recipes.LatestRecipes' | translate }}</h2>
 		</div>
 	</div>
 </div>
+
+<div
+	class="d-flex justify-content-center mt-3"
+	*ngIf="(recipesAreLoaded$ | async) === false">
+	<div class="spinner-border text-primary" role="status">
+		<span class="visually-hidden">Loading...</span>
```
Comment (danieljanusm) [inline src/app/recipes/recipes-cards/recipes-cards.component.html:18]: use i18n

File: src/app/recipes/recipes-cards/recipes-cards.component.ts [hunk 3]
```diff
@@ -3,16 +3,20 @@ import { Store } from '@ngrx/store';
...
-	public recipes$: Observable<Recipe[]> = this.store.select(
-		RecipesSelectors.selectRecipes
-	);
-	constructor(private store: Store<AppState>) {}
+	public recipes$: Observable<Recipe[]>;
+	public recipesAreLoaded$: Observable<boolean>;
+	constructor(private store: Store<AppState>) {
+		this.recipes$ = this.store.select(RecipesSelectors.selectRecipes);
+		this.recipesAreLoaded$ = this.store.select(
+			RecipesSelectors.selectRecipesAreLoaded
```
Comment (danieljanusm) [inline src/app/recipes/recipes-cards/recipes-cards.component.ts:19]: move all this code from constructor to ngOnInit

https://javascript.plainenglish.io/difference-between-constructor-and-ngoninit-in-angular-537ecfa6ce1e

File: src/app/store/recipes/effects/load-success-recipe.effect.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+@Injectable()
+export class loadRecipesSuccessEffect {
+	loadRecipes$ = createEffect(() => {
+		return this.actions$.pipe(
+			ofType(RecipesActions.loadRecipesSuccess),
+			map(action => {
+				const { recipe } = action;
+				switch (action.action) {
+					case 'added':
```
Comment (danieljanusm) [inline src/app/store/recipes/effects/load-success-recipe.effect.ts:16]: declare an Enum and use it here

Added = 'added' etc.

File: src/app/store/recipes/recipes.actions.ts [hunk 5]
```diff
@@ -1,15 +1,44 @@
...
+);
+
+const firestoreModifyRecipe = createAction(
+	'[Firestore] Recipe modified',
+	props<{ recipe: Recipe }>()
+);
+
+const firebaseRemoveRecipe = createAction(
+	'[Firestore] Recipe removed',
+	props<{ recipe: Recipe }>()
```
Comment (danieljanusm) [inline src/app/store/recipes/recipes.actions.ts:27]: id is not enough here?

**Review by danieljanusm — CHANGES_REQUESTED** [review]

File: src/app/store/recipes/effects/load-recipes-effect.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+					id: action.payload.doc.id,
+				};
+				return RecipesActions.loadRecipesSuccess({
+					action: action.type,
+					recipe,
+				});
+			})
+		);
+	});
+	constructor(private actions$: Actions, private afs: AngularFirestore) {}
```
Comment (Ciszak-Krzysztof) [inline src/app/store/recipes/effects/load-recipes-effect.ts:29]: why so big tabs?

File: src/app/store/recipes/effects/load-success-recipe.effect.ts (hunk 4 above)
Comment (jakubjen) [reply src/app/store/recipes/effects/load-success-recipe.effect.ts:16]: I use DocumentChangeType so is type save and vscode autocompletion work

File: src/app/store/recipes/recipes.actions.ts (hunk 5 above)
Comment (jakubjen) [reply src/app/store/recipes/recipes.actions.ts:27]: I change that

File: src/app/store/recipes/effects/load-recipes-effect.ts (hunk 6 above)
Comment (jakubjen) [reply src/app/store/recipes/effects/load-recipes-effect.ts:29]: I use tabs instead spaces in prettier config

**Pull request merged**